proposed/activity-improvements.md

```diff
@@ -0,0 +1,465 @@
+  When calling `ActivitySource.AddActivityListener()`, the provided callback `ListenToSource(activitySource)` will get called for every ActivitySource previously created and still active to decide if need to listen to Activities created from such ActivitySource. Also, this callback will get called for every future created ActivitySource.
+
+Here is example of how to create a listener for Activities events.
+
+```C#
+ActivityListener listener = new ActivityListener
+{
+    ActivityStopped = (activity) => { ... },
+    ActivityStarted = (activity) => { ... },
+    ShouldListenTo = (activitySource) => ...,
+    GetRequestedDataUsingParentId = (ref ActivityCreationOptions<string> options) => ...,
+    GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> options) => ... 
+};
+
+ActivitySource.AddActivityListener(listener);
+
+...
+    
+listener.Dispose();  
+```
+
+## Proposed API
+
+### ActivityKind
+
+```C#
+namespace System.Diagnostics
+{
+    // Kind describes the relationship between the Activity, its parents, and its children in a Trace.
+    public enum ActivityKind
+    {
+        // Indicates that the Activity represents an internal operation within an application, as opposed to an operation with remote parents or children.
+        Internal = 0,
+
+        /// Server activity represents request incoming from external component.
+        Server = 1,
+
+        /// Client activity represents outgoing request to the external component.
+        Client = 2,
+
+        /// Producer activity represents output provided to external components.
+        Producer = 3,
+
+        /// Consumer activity represents output received from an external component.
+        Consumer = 4,
+    }
+}
+```
+
+ ### ActivityContext
+
+```C#
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// ActivityContext representation conforms to the w3c TraceContext specification. It contains two identifiers
+    /// a TraceId and a SpanId - along with a set of common TraceFlags and system-specific TraceState values.
+    /// </summary>
+    public readonly struct ActivityContext : IEquatable<ActivityContext>
```

Comment:
Is this intended to replace SpanContext? (https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#spancontext)

`IsRemote` is a property in SpanContext OT spec, but omitted here. Is it intentional?

Reply:
> Is this intended to replace SpanContext?

Yes.

> IsRemote is a property in SpanContext OT spec but omitted here. Is it intentional?

Yes, this was discussed before and the feedback is most likely IsRemote is going to be removed from OT. That is why we didn't add it to BCL.